Services/TodoService.cs

```diff
@@ -0,0 +1,85 @@
+using System;
+using Xunit;
+using Moq;
+using TodoApp.Repositories.Interfaces;
+using TodoApp.Repositories.XmlRepository;
+using Service = TodoApp.Services;
+using TodoApp.Repositories.XmlRepository.Utils;
+using System.Xml.Linq;
+using System.Linq;
+using System.Globalization;
+
+namespace TodoApp.Tests.Services
+{
+    public class TodoService
+    {
+
+        protected Mock<ITodoRepository> MockRepository;
+
+        IXmlContext context = new XmlContext("");
+
+        protected XElement Container;
+
+        public TodoService()
+        {
+            MockRepository = new Mock<ITodoRepository>();
+
+            string containerName = "todos";
+            Container = context.GetContainer(containerName);
+
+     
+            var firstTodo = new XElement("todo");
+            firstTodo.Add(new XAttribute("Id", "00000000-0000-0000-0000-000000000001"));
+            firstTodo.Add(new XElement("Title", "Unit tests"));
+            firstTodo.Add(new XElement("Description", "Learn how to make unit tests"));
+            firstTodo.Add(new XElement("Status", "Open"));
+            firstTodo.Add(new XElement("CreatedOn", "2020-04-15T14:29:15.1823029Z"));
+            firstTodo.Add(new XElement("DueDate", "2020-04-19T21:00:00.0000000Z"));
+
+            var secondTodo = new XElement("todo");
+            secondTodo.Add(new XAttribute("Id", "20975aeb-d490-4aa6-95ba-5b7c50b074a4"));
+            secondTodo.Add(new XElement("Title", "Football"));
+            secondTodo.Add(new XElement("Description", "Play football"));
+            secondTodo.Add(new XElement("Status", "Open"));
+            secondTodo.Add(new XElement("CreatedOn", "2020-04-16T10:17:33.2653554Z"));
+            secondTodo.Add(new XElement("DueDate", "2020-05-29T21:00:00.0000000Z"));
+
+            var thirdTodo = new XElement("todo");
+            thirdTodo.Add(new XAttribute("Id", "30000300-4000-0000-0900-00a0000f0050"));
+            thirdTodo.Add(new XElement("Title", "Another todo"));
+            thirdTodo.Add(new XElement("Description", ""));
+            thirdTodo.Add(new XElement("Status", "InProgress"));
+            thirdTodo.Add(new XElement("CreatedOn", "2020-09-20T14:29:15.1823029Z"));
+            thirdTodo.Add(new XElement("DueDate", "2020-09-30T21:00:00.0000000Z"));
+
+            Container.Add(firstTodo);
+            Container.Add(secondTodo);
+
+
+            ITodoRepository repo = new TodoRepository(context);
+        }
+
+        [Fact]
+        public void GetAll_ReturnsAllTodos()
```

Comment:
In unit testing, you need to mock all dependencies of the code under test.

In this case you are testing the service, more specifically `GetAll()` method.
The only dependency of the service is `ITodoRepository`. No XML, and no context.
You need to mock the methods that are executed.
`service.GetAll()` just calls `repository.GetAll()`, so we need to mock only that method for this test.

```cs
var mockTodos = new TodoModel[]
{
    new TodoModel()
    {
        Id = new Guid(),
        Title = "Test",
        Description = "Test Description",
        Status = TodoStatus.Open,
        DueDate = new DateTime(2021, 3, 4, 12, 30, 00),
        CreatedOn = new DateTime(2020, 11, 12, 11, 55, 13),
    },
    new TodoModel()
    {
        Id = new Guid(),
        Title = "Second",
        Description = "Second Test",
        Status = TodoStatus.InProgress,
        DueDate = new DateTime(2021, 1, 24, 1, 22, 33),
        CreatedOn = new DateTime(2020, 10, 11, 12, 13, 14),
    },
};
MockRepository.Setup(a => a.GetAll()).Returns(mockTodos);
var service = new Service.TodoService(MockRepository.Object);

var todos = service.GetAll();

Assert.Equal(mockTodos, todos);
MockRepository.Verify(a => a.GetAll(), Times.Once);
```

Note: We have 2 assertions, so we would want to split this into two different tests: `GetAll_ReturnsAllTodos` and `GetAll_CallRepositoryGetAll`